odo service create panic when there is no Service Catalog available on cluster

[kind/bug]

<!--

Welcome! - We kindly ask you to:

  1. Fill out the issue template below 
  2. Use the chat and talk to us if you have a question rather than a bug or feature request.

The chat room is at: https://chat.openshift.io/developers/channels/odo

Thanks for understanding, and for contributing to the project!

-->

## What versions of software are you using?
- Operating System:
- Output of `odo version`: current master -  v0.0.18 (c25f00ef)


## Actual behavior

1. Connect to the cluster that does't have Service Catalog deployed.
2. `odo service create`

```
▶ odo service create
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1ccd6ff]

goroutine 1 [running]:
github.com/redhat-developer/odo/pkg/service.NewServicePlan(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/pkg/service/service.go:353 +0xef
github.com/redhat-developer/odo/pkg/odo/cli/service/ui.enterServicePropertiesInteractively(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/pkg/odo/cli/service/ui/ui.go:259 +0x92
github.com/redhat-developer/odo/pkg/odo/cli/service/ui.EnterServicePropertiesInteractively(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/pkg/odo/cli/service/ui/ui.go:253 +0x76
github.com/redhat-developer/odo/pkg/odo/cli/service.(*ServiceCreateOptions).Complete(0xc000bba6c0, 0x2485701, 0x6, 0xc000660900, 0x376fc58, 0x0, 0x0, 0xc000b6dc40, 0x0)
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/pkg/odo/cli/service/create.go:90 +0x434
github.com/redhat-developer/odo/pkg/odo/cli/service.NewCmdServiceCreate.func1(0xc000660900, 0x376fc58, 0x0, 0x0)
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/pkg/odo/cli/service/create.go:188 +0x6d
github.com/redhat-developer/odo/vendor/github.com/spf13/cobra.(*Command).execute(0xc000660900, 0x376fc58, 0x0, 0x0, 0xc000660900, 0x376fc58)
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:702 +0x2d3
github.com/redhat-developer/odo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000678000, 0x8, 0xc000000300, 0x20b1a82)
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:783 +0x2dc
github.com/redhat-developer/odo/vendor/github.com/spf13/cobra.(*Command).Execute(0xc000678000, 0x2585f38, 0xc000706600)
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/Users/tkral/Code/goenv/odo/src/github.com/redhat-developer/odo/cmd/odo/odo.go:64 +0x350
```

## Expected behavior

odo should show error message explaining that cluster that user is connected to doesn't support Service Catalog.

--- a/pkg/occlient/occlient.go
+++ b/pkg/occlient/occlient.go
@@ -2234,7 +2234,7 @@ func (c *Client) GetServiceClassesByCategory() (categories map[string][]scv1beta
 	categories = make(map[string][]scv1beta1.ClusterServiceClass)
 	classes, err := c.GetClusterServiceClasses()
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to get cluster service classes")
+		return nil, err
 	}
 
 	// TODO: Should we replicate the classification performed in

--- a/pkg/odo/cli/service/create.go
+++ b/pkg/odo/cli/service/create.go
@@ -72,13 +72,13 @@ func (o *ServiceCreateOptions) Complete(name string, cmd *cobra.Command, args []
 	if o.interactive {
 		classesByCategory, err := client.GetServiceClassesByCategory()
 		if err != nil {
-			return errors.Wrap(err, "couldn't retrieve the service classes")
+			return fmt.Errorf("unable to retrieve service classes: %v", err)
 		}
 		class, o.serviceType = ui.SelectClassInteractively(classesByCategory)
 
 		plans, err := client.GetMatchingPlans(class)
 		if err != nil {
-			return errors.Wrap(err, fmt.Sprintf("couldn't retrieve plans for class %s", class.GetExternalName()))
+			return fmt.Errorf("couldn't retrieve plans for class %s: %v", class.GetExternalName(), err)
 		}
 
 		var svcPlan scv1beta1.ClusterServicePlan

--- a/pkg/odo/cli/service/create.go
+++ b/pkg/odo/cli/service/create.go
@@ -2,6 +2,7 @@ package service
 
 import (
 	"fmt"
+	"github.com/pkg/errors"
 	"github.com/redhat-developer/odo/pkg/odo/util/validation"
 	"strings"
 
@@ -69,10 +70,16 @@ func (o *ServiceCreateOptions) Complete(name string, cmd *cobra.Command, args []
 
 	var class scv1beta1.ClusterServiceClass
 	if o.interactive {
-		classesByCategory, _ := client.GetServiceClassesByCategory()
+		classesByCategory, err := client.GetServiceClassesByCategory()
+		if err != nil {
+			return errors.Wrap(err, "couldn't retrieve the service classes")
+		}
 		class, o.serviceType = ui.SelectClassInteractively(classesByCategory)
 
-		plans, _ := client.GetMatchingPlans(class)
+		plans, err := client.GetMatchingPlans(class)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("couldn't retrieve plans for class %s", class.GetExternalName()))
+		}
 
 		var svcPlan scv1beta1.ClusterServicePlan
 		// if there is only one available plan, we select it
@@ -130,7 +137,7 @@ func (o *ServiceCreateOptions) Validate() (err error) {
 	// make sure the service type exists
 	classPtr, err := o.Client.GetClusterServiceClass(o.serviceType)
 	if err != nil {
-		return fmt.Errorf("unable to create service because Service Catalog is not enabled in your cluster:\n%v", err)
+		return errors.Wrap(err, "unable to create service because Service Catalog is not enabled in your cluster")
 	}
 	if classPtr == nil {
 		return fmt.Errorf("service %v doesn't exist\nRun 'odo catalog list services' to see a list of supported services.\n", o.serviceType)

